lib/vrl/lookup/src/lookup_v2/mod.rs

```diff
@@ -1,51 +1,83 @@
 //! when passing in a value (e.g. [OwnedValuePath]) into a function that requires `impl ValuePath`,
-//! you will generally need to pass it in as a reference.
+//! it will generally need to be passed in as a reference.
 //!
 //! # Owned Paths
-//! [OwnedValuePath] and [OwnedTargetPath] are fully parsed paths. These contain heap-allocated
-//! segments.
+//! [OwnedValuePath] and [OwnedTargetPath] are pre-parsed paths. That means that accessing fields
+//! using an owned path is very fast. There is an upfront cost however, since owned paths are parsed
+//! when they are created, and the segments are heap allocated. Owned paths should be preferred
+//! when they can be created when performance isn't as much of a concern (e.g. startup time)
+//! and it can be stored to use multiple times.
+//! Owned paths tend to be easier to work with since you can directly access / manipulate the
+//! segments that make up the path.
 //!
-//! Owned paths should generally be preferred over other types (such as strings) whenever feasible
-//! However since owned paths are allocated / pre-parsed, if a path is being created and will
-//! only be used once, it may make sense to use other types. For example, if you have an
-//! [OwnedValuePath] and want to append a segment to it before querying a `Value` you could:
+//! If a path is being created and will only be used once, it may make sense to use other types.
+//! For example here are two different ways to append a segment to a [OwnedValuePath]  before querying
+//! a `Value`:
 //! - Use [OwnedValuePath::with_field_appended] to create a new [OwnedValuePath] and use that. This
 //! method is preferred if the new path will be used multiple times and the path adjustment can be
 //! done in a non performance-critical part of the code (e.g. at startup).
 //! - Use [ValuePath::concat] which con concatenate two [ValuePath]'s very efficiently without
 //! allocating on the heap.
 //!
-//! To convert a string into an owned path, you can use either [parse_value_path] or [parse_target_path].
+//! To convert a string into an owned path, use either [parse_value_path] or [parse_target_path].
 //!
 //! # String Paths
 //! [ValuePath] and [TargetPath] are implemented for [&str]. That means a raw / unparsed string can
 //! be used as a path. This use is discouraged, and may be removed in the future. It mostly
 //! exists for backwards compatibility in places where String paths are used instead of owned paths.
-//! Using string paths is still very performant, but it's easy to introduce bugs since some type
-//! information is missing, such as whether it's a target vs value path, or if the entire string
-//! is meant to be treated as a single segment vs being parsed as a path.
+//! Using string paths is slightly slower than using an owned path (it's still very fast),
+//! but it's easy to introduce bugs since some type information is missing, such as whether it's a
```

Comment:
could be helpful to qualify "type" as like Rust type or compiled type.

Otherwise this could be read as "some type of information is missing" by brains not thinking about type systems all the time.